### PR TITLE
Update kubernetes api demo to use kubernetes api

### PR DESCRIPTION
**Pull request by chbatey**
* Remove DNS from docs
* Remove headless service as it was only required when using dns with A
records

### CODE REVIEW

File: bootstrap-demo/kubernetes-api/build.sbt [hunk 1]
```diff
@@ -2,37 +2,22 @@ import com.typesafe.sbt.packager.docker._
...
-*/
+version := "1.3.3.7" // we hard-code the version here, it could be anything really
 
 dockerCommands :=
   dockerCommands.value.flatMap {
     case ExecCmd("ENTRYPOINT", args @ _*) => Seq(Cmd("ENTRYPOINT", args.mkString(" ")))
     case v => Seq(v)
   }
+dockerExposedPorts := Seq(8080, 8558, 2552)
+dockerBaseImage := "openjdk:8-jre-alpine"
```
Comment (chbatey) [inline bootstrap-demo/kubernetes-api/build.sbt]: Reduces image size by 800 megs

File: bootstrap-demo/kubernetes-api/kubernetes/akka-cluster.yml [hunk 2]
```diff
@@ -42,35 +36,6 @@ spec:
           containerPort: 8558
           protocol: TCP
 ---
```
Comment (chbatey) [inline bootstrap-demo/kubernetes-api/kubernetes/akka-cluster.yml]: We don't need this when using the kubernetes api

File: bootstrap-demo/kubernetes-api/src/main/resources/application.conf [hunk 3]
```diff
@@ -1,38 +1,27 @@
...
-        service-name = "appka-service"
-        service-namespace = "default.svc.cluster.local"
-        stable-margin = 5 seconds
-      }
-
-      contact-point {
-        # currently this port HAS TO be the same as the `akka.management.http.port`
-        # it would not have to be once we implement the SRV record watching, since then we could potentially
-        # get the ports from the DNS records.
-        fallback-port = 8558
```
Comment (chbatey) [inline bootstrap-demo/kubernetes-api/src/main/resources/application.conf]: k8s api gives us the port

File: docs/src/main/paradox/bootstrap/demos/index.md [hunk 4]
```diff
@@ -1,7 +1,19 @@
-# Bootstrap examples 
+# Bootstrap recipes
```
Comment (chbatey) [inline docs/src/main/paradox/bootstrap/demos/index.md]: I made these two guides more consistent but they still need some more work

**Comment by chbatey** [comment]
Closes #292

File: bootstrap-demo/kubernetes-api/kube-apply.sh [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+kubectl apply -f kubernetes/akka-cluster.yml
```
Comment (jrudolph) [inline bootstrap-demo/kubernetes-api/kube-apply.sh]: Is that script still useful? Just showing the `kubectl apply` command seems more explicit and should be understandable by the intended audience (if that's something you can say about `kubectl apply` at all--when you read around the web, `kubectl apply` seems like the "starting exe files sent to you by strangers via email" of the modern systems administrator... ;) ).

File: bootstrap-demo/kubernetes-api/build.sbt [hunk 6]
```diff
@@ -2,37 +2,22 @@ import com.typesafe.sbt.packager.docker._
...
-libraryDependencies += Seq(
-  // 1) we want to use akka-management
-  "akka-management",
-
-  // 2) we want to use the bootstrap
-  "akka-management-cluster-bootstrap",
-
-  // 3) the initial contact points should be located via DNS:
-  "akka-discovery-dns",
+//dockerEntrypoint ++= Seq(
```
[the remark this answers is not in the document]
Comment (chbatey) [reply bootstrap-demo/kubernetes-api/build.sbt]: Nope

[the remark this answers is not in the document]
Comment (chbatey) [reply bootstrap-demo/kubernetes-api/build.sbt]: Nope

File: bootstrap-demo/kubernetes-api/kubernetes/akka-cluster.yml [hunk 7]
```diff
@@ -17,14 +17,8 @@ spec:
     spec:
       containers:
       - name: appka
-        image: ktoso/akka-management-bootstrap-joining-demo-kubernetes-api:1.3.3.7
+        image: bootstrap-demo-kubernetes-api:1.3.3.7
```
[the remark this answers is not in the document]
Comment (chbatey) [reply bootstrap-demo/kubernetes-api/kubernetes/akka-cluster.yml]: Yep

File: bootstrap-demo/kubernetes-api/src/main/resources/application.conf [hunk 8]
```diff
@@ -1,38 +1,27 @@
...
 
-  discovery {
-    # pick the discovery method you'd like to use:
-    method = kubernetes-api
-
-    kubernetes-api {
-      pod-label-selector = "actorSystemName=appka"
-    }
+#management-config
+akk.management {
```
[the remark this answers is not in the document]
Comment (chbatey) [reply bootstrap-demo/kubernetes-api/src/main/resources/application.conf]: This is the same as the reference conf 😀 will just remove entirely

[the remark this answers is not in the document]
Comment (chbatey) [reply bootstrap-demo/kubernetes-api/src/main/resources/application.conf]: And the other stuff in there was ignored as there wasn't a %s in the pod label selector for the kubernetes api. I'll update it to use the kubernetes-api as intended

File: docs/src/main/paradox/bootstrap/demos/index.md [hunk 9]
```diff
@@ -1,7 +1,19 @@
...
 ## Kubernetes
 
-For Akka Cluster in Kubernetes either `akka-dns` or `kubernetes-api` can be used:
+For Akka Cluster in Kubernetes either `akka-dns` or `kubernetes-api` service discovery can be used:
+
 * @ref[Kubernetes using `akka-dns` discovery](./kubernetes.md)
 * @ref[Kubernetes using `kubernetes-api` discovery](./kubernetes-api.md)
+
+DNS has the benefit that it is agnostic of Kubernetes so does not require pods be able to communicate with the API
+server. However it requires a headless service that supports the `publishNotReadyAddresses` feature.
```
[the remark this answers is not in the document]
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/demos/index.md]: I'll add a note to say if your k8s setup doesn't support this then kubernetes api should be used

File: docs/src/main/paradox/bootstrap/demos/kubernetes-api.md [hunk 10]
```diff
@@ -1,20 +1,26 @@
...
 
-An example application using docker and prepared to be deployed to kubernetes is provided in Akka Management's github repository 
-as sub-project [bootstrap-joining-demo](https://github.com/akka/akka-management/tree/master/bootstrap-joining-demo/kubernetes-api).
+The following Kubernetes resources are created:
 
-Rather than configuring the Dockerfile directly, we used the [sbt-native-packager](http://sbt-native-packager.readthedocs.io/en/stable/) 
-to package the application as docker container. See the `build.sbt` file for more details, and the `kubernetes/akka-cluster.yml` 
-file for the service configuration, which is:
+* Deployment: For creating the Akka pods 
+* Role and RoleBinding to give the pods access to the API server
```
[the remark this answers is not in the document]
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/demos/kubernetes-api.md]: I am not sure. It allows read only access to pod names

File: bootstrap-demo/kubernetes-api/kube-apply.sh (hunk 5 above)
Comment (chbatey) [reply bootstrap-demo/kubernetes-api/kube-apply.sh]: Yep, will remove these and just add the command

File: bootstrap-demo/kubernetes-api/src/main/resources/application.conf [hunk 11]
```diff
@@ -1,38 +1,21 @@
...
-        fallback-port = 8558
-      }
+}
+#discovery-config
+
+#management-config
+akka.management {
+  cluster.bootstrap {
+    contact-point-discovery {
+      # For the kubernetes API is is substributed into the %s in pod-label-selector
```
Comment (raboof) [inline bootstrap-demo/kubernetes-api/src/main/resources/application.conf]: 'it'?

Comment (raboof) [reply bootstrap-demo/kubernetes-api/src/main/resources/application.conf]: or perhaps 'this value'

File: docs/src/main/paradox/bootstrap/demos/index.md [hunk 12]
```diff
@@ -1,7 +1,20 @@
-# Bootstrap examples 
+# Bootstrap recipes 
+
+A set of bootstrap demonstration projects can be found in [bootstrap-demo](https://github.com/akka/akka-management/tree/master/bootstrap-demo) folder of this project. Currently there are projects for:
+
+* AWS API
+* Mesos using DNS
+* Kubernetes using the API server
+* Kubernetes using DNS
+* Marathon
```
Comment (raboof) [inline docs/src/main/paradox/bootstrap/demos/index.md]: Maybe mention K8s first?

File: docs/src/main/paradox/bootstrap/demos/kubernetes.md [hunk 13]
```diff
@@ -1,9 +1,9 @@
-# Kubernetes
+# Kubernetes using DNS
 
-A full working example that can be deployed to kubernetes via `minikube` is in `bootstrap-demo/kubernetes-dns`.
+An example project that can be deployed to kubernetes via `minikube` is in `bootstrap-demo/kubernetes-dns`.
 
-As of Akka 2.5.15 and Akka-Management 0.18 `akka-dns` discovery mechanism can be used to deploy Akka Cluster in Kubernetes.
-This example shows how to:
+As of Akka 2.5.15 and Akka-Management 0.18 the `akka-dns` discovery mechanism can be used to deploy Akka Cluster in Kubernetes.
```
Comment (raboof) [inline docs/src/main/paradox/bootstrap/demos/kubernetes.md]: Should `akka-dns` be changed to `async-dns` now?

Comment (raboof) [reply docs/src/main/paradox/bootstrap/demos/kubernetes.md]: We'll want to remove the 'As of Akka 2.5.15 and Akka-Management 0.18' at some point, but might be useful for now.

File: docs/src/main/paradox/bootstrap/demos/kubernetes-api.md [hunk 14]
```diff
@@ -42,54 +48,41 @@ $ eval $(minikube docker-env)
...
-[info] Built image ktoso/akka-management-bootstrap-joining-demo-kubernetes-api:1.3.3.7
-[success] Total time: 25 s, completed Dec 8, 2017 7:47:05 PM
 ```
 
-Once the image is published, you can deploy it onto the kubernetes cluster by calling:
+Once the image is published, deploy it onto the kubernetes cluster:
 
-@@snip [kube-create.sh]($management$/bootstrap-demo/kubernetes-api/kube-create.sh) 
+```
+kubectl apply -f kuberbetes/akka-cluster.yml
```
Comment (raboof) [inline docs/src/main/paradox/bootstrap/demos/kubernetes-api.md]: Do we also document how to clean up the cluster?

File: docs/src/main/paradox/bootstrap/demos/kubernetes-api.md [hunk 15]
```diff
@@ -1,20 +1,26 @@
...
-You run the example using [minikube](https://github.com/kubernetes/minikube) (or a real kubernetes system),
-which you can do by typing:
+The following configuration is required:
+
+* Set `akka.discovery.method` to `kubernetes-api`
+* Set `akka.discovery.kubernetes-api.pod-label-selector` to a label selector that will match the Akka pods
+
+@@snip [akka-cluster.yml]($management$/bootstrap-demo/kubernetes-api/src/main/resources/application.conf) { #discovery-config } 
+
+You can run the example using [minikube](https://github.com/kubernetes/minikube) (or a real Kubernetes system) by:
```
Comment (raboof) [inline docs/src/main/paradox/bootstrap/demos/kubernetes-api.md]: The instructions below expect minikube though. Perhaps split this up in "set up K8s, for example like this with Minikube or another K8s system" and then pick up at step 4 for 'general' instructions again?

File: docs/src/main/paradox/bootstrap/demos/kubernetes.md (hunk 13 above)
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/demos/kubernetes.md]: The service discovery mechanism is `akka-dns` which uses the Akka `async-dns` DNS provider

Comment (raboof) [reply docs/src/main/paradox/bootstrap/demos/kubernetes.md]: gotcha :+1:

File: docs/src/main/paradox/bootstrap/demos/index.md (hunk 12 above)
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/demos/index.md]: +1

File: docs/src/main/paradox/bootstrap/demos/kubernetes.md (hunk 13 above)
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/demos/kubernetes.md]: I think we can remove it now. We have an explicit check for akka 2.5.15

File: docs/src/main/paradox/bootstrap/demos/kubernetes-api.md (hunk 15 above)
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/demos/kubernetes-api.md]: :+1:

File: docs/src/main/paradox/bootstrap/demos/kubernetes-api.md (hunk 14 above)
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/demos/kubernetes-api.md]: Will add

**Pull request merged**